addons/mail/static/tests/chat_window/chat_window_tests.js

```diff
@@ -404,6 +404,25 @@ QUnit.test("Close active thread action in chatwindow on ESCAPE", async () => {
     await contains(".o-mail-ChatWindow");
 });
 
+QUnit.test("Exit chatwindow thread rename on ESCAPE without closing chat window", async () => {
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create({
+        name: "General",
+        channel_member_ids: [
+            Command.create({ partner_id: pyEnv.currentPartnerId, is_minimized: true }),
+        ],
+        create_uid: pyEnv.currentUserId,
+    });
+    await start();
+    await contains(".o-mail-ChatWindow");
+    await click(".o-mail-ChatWindow-command", { text: "General" });
+    await click(".dropdown-item", { text: "Rename" });
+    await contains(".o-mail-AutoresizeInput.o-focused");
+    triggerHotkey("Escape");
+    await contains(".o-mail-AutoresizeInput.o-focused", { count: 0 });
+    await contains(".o-mail-ChatWindow");
```

Comment:
I know the fix at hand is not closing chat window while cancelling the rename, but the test core aspect for improved maintainability is to test that cancelling rename in chat window works (implicitly tests that chat window is still present).

Therefore should actually put a name and ESC to `contains()` that name is effectively canceled on chat window header.